recipe-client-addon/lib/PreferenceExperiments.jsm

```diff
@@ -186,10 +187,39 @@ this.PreferenceExperiments = {
       lastSeen: new Date().toJSON(),
       preferenceName,
       preferenceValue,
+      preferenceType,
       previousPreferenceValue: preferences.get(preferenceName, undefined),
       preferenceBranchType,
     };
 
+    // If there's a previous preference value set, we need to ensure that the
+    // incoming value is of the same type.
+    if (preferences.has(preferenceName)) {
```

Comment:
My understanding is that if:

1) a pref experiment uses a user-branch pref
2) there's a default branch value

this code won't detect it. Is that intentional? (I don't think it should be, but it's possible I'm missing something...)